ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/InjectTxs.hs

```diff
@@ -7,11 +7,18 @@
 {-# LANGUAGE TypeOperators       #-}
 
 module Ouroboros.Consensus.HardFork.Combinator.InjectTxs (
+    -- * Transactions before validation
     InjectTx (..)
   , cannotInjectTx
   , matchTx
   , matchTxNS
   , matchTxsNS
+    -- * Transactions after validation
```

Comment:
This is a concerning amount of duplication. I imagine we could generalize the existing code to handle both `GenTx` and `Validated (GenTx blk)`, but I at least wanted to get everything compiling before thinking about how to generalize.

I'm also concerned that the genearlized code may be more dense. So we should study this trade-off here.

Reply:
I've pushed up some additional commits that remove this duplication by instead generalizing the definition. We need to discuss whether the additional abstraction is worthwhile. I'm feeling that it is -- but I just wrote, so I may be underestimating it.

There are also other options, such as introducing a new GADT as a parameter or introducing type classes etc instead of abstracting out the type-specific parts as HOF arguments.

Reply:
I think this is fine. `GenTx` and `ValidatedGenTx` are "essentially" the same thing with some annotations on the latter so it makes sense they are generalized in a common concept that is used to inject them and so. For me this looks good.

Reply:
Great, Resolving.